### DISCUSSION TITLE
Pirania captive portal only for wlan0-ap

### DISCUSSION THREAD

**Discussion by germanferrero** [libremesh/lime-packages#905] [General]
Hi @altergui, @G10h4ck, @luandro, @spiccinini  and everyone that may have a clue of how to do this...

I'm trying to make pirania's captive portal to work only for clients connected to the wlan0-ap interface.
I've tried with `-i` option for iptables. It does work for br-lan and anygw (as configured by default) but doesn't seem to work for bridged interfaces (wlan0-apname, wlan0-ap, etc).

My next step is to try out [iptables-mod-physdev](https://openwrt.org/packages/pkgdata/iptables-mod-physdev) but `phys` makes me think of radio0, radio1, and such.

I think I need some conceptual clarity to address this task.
Could you help me?

Thanks!

**Comment by germanferrero** [comment]
So I need to make iptables filter traffic for an specific `wireless operating mode virtual interfaces`. Source: https://openwrt.org/docs/guide-developer/networking/network.interfaces

**Comment by ilario** [comment]
Maybe ebtables can?

**Reply by ilario** [reply]
https://serverfault.com/a/607814/508349

**Reply by germanferrero** [reply]
Thanks @ilario. I was able to drop all packets coming through wlan0-ap with:
`ebtables -A INPUT -i wlan0-ap -j DROP`
That's a step forward. I'll be trying to find the way to integrate this with the rest of the captive-portal script

**Reply by germanferrero** [reply]
I got a working implementation and requested for comments on: https://github.com/spiccinini/lime-packages/pull/9
